google_cloud_automlops/orchestration/kfp/KFPComponent.py

```diff
@@ -0,0 +1,210 @@
+
+# pylint: disable=anomalous-backslash-in-string
+# pylint: disable=C0103
+# pylint: disable=line-too-long
+
+from typing import Callable, List, Optional
+
+try:
+    from importlib.resources import files as import_files
+except ImportError:
+    # Try backported to PY<37 `importlib_resources`
+    from importlib_resources import files as import_files
+
+from google_cloud_automlops.orchestration.Component import Component
+from google_cloud_automlops.utils.constants import (
+    BASE_DIR,
+    GENERATED_DEFAULTS_FILE,
+    GENERATED_LICENSE,
+    KFP_TEMPLATES_PATH,
+    PLACEHOLDER_IMAGE,
+)
+from google_cloud_automlops.utils.utils import (
+    make_dirs,
+    read_yaml_file,
+    render_jinja,
+    write_file,
+    write_yaml_file
+)
+
+
+class KFPComponent(Component):
+    """Creates a KFP specific Component object for #TODO: add more
+
+    Args:
+        Component (object): Generic Component object.
+    """
+
+    def __init__(self,
+                 func: Optional[Callable] = None, 
+                 packages_to_install: Optional[List[str]] = None):
+        """Initiates a KFP Component object created out of a function holding
+        all necessary code.
+
+        Args:
+            func: The python function to create a component from. The function
+                should have type annotations for all its arguments, indicating how
+                it is intended to be used (e.g. as an input/output Artifact object,
+                a plain parameter, or a path to a file).
+            packages_to_install: A list of optional packages to install before
+                executing func. These will always be installed at component runtime.
+        """
+        super().__init__(func, packages_to_install)
+
+        # Update parameters and return types to reflect KFP data types
+        if self.parameters:
+            self.parameters = update_params(self.parameters)
+        if self.return_types:
+            self.return_types = update_params(self.return_types)
+
+        # Set packages to install and component spec attributes
+        self.packages_to_install_command = self._get_packages_to_install_command()
+        self.component_spec = self._create_component_spec()
+
+    def build(self):
+        """Constructs files for running and managing Kubeflow pipelines.
+        """
+        defaults = read_yaml_file(GENERATED_DEFAULTS_FILE)
+        self.artifact_repo_location = defaults['gcp']['artifact_repo_location']
+        self.artifact_repo_name = defaults['gcp']['artifact_repo_name']
+        self.project_id = defaults['gcp']['project_id']
+        self.naming_prefix = defaults['gcp']['naming_prefix']
+
+        # Set and create directory for components if it does not already exist
+        component_dir = BASE_DIR + 'components/' + self.component_spec['name']
+
+        # Build necessary folders
+        # TODO: make this only happen for the first component? or pull into automlops.py
+        make_dirs([
```

Comment:
This should be one of the very first things that runs when KFP is selected as the orchestration framework. Can a pipeline exist without components? If not, I think it makes sense to keep this here. Also, if other orchestration frameworks require different directory styles, make_dirs function should be moved from utils to a method or util within the KFP classes.